mslice/plotting/plot_window/slice_plot.py

```diff
@@ -213,27 +213,37 @@ def _run_temp_dependent(self, slice_plotter_method, previous):
             slice_plotter_method(self._ws_title)
         except ValueError:  # sample temperature not yet set
             try:
-                field = self.ask_sample_temperature_field(str(self._ws_title))
+                temp_value, field = self.ask_sample_temperature_field(str(self._ws_title))
             except RuntimeError:  # if cancel is clicked, go back to previous selection
                 self.intensity_selection(previous)
                 return False
-            self._slice_plotter.add_sample_temperature_field(field)
-            self._slice_plotter.update_sample_temperature(self._ws_title)
+            if field:
+                self._slice_plotter.add_sample_temperature_field(temp_value)
+                self._slice_plotter.update_sample_temperature(self._ws_title)
+            else:
+                try:
+                    temp_value = float(temp_value)
+                except ValueError:
+                    self.plot_figure.error_box("Invalid value entered for sample temperature. Enter a number or a \
+                                               sample log field.")
+                    self.intensity_selection(previous)
+                    return False
+                else:
+                    self._slice_plotter.set_sample_temperature(self._ws_title, temp_value)
             slice_plotter_method(self._ws_title)
         return True
 
     def ask_sample_temperature_field(self, ws_name):
-        if ws_name[-3:] == '_QE':
-            ws_name = ws_name[:-3]
+        ws_name = ws_name[:ws_name.rfind("_")]
         ws = AnalysisDataService[ws_name]
         temp_field, confirm = QtWidgets.QInputDialog.getItem(self.plot_figure, 'Sample Temperature',
                                                              'Sample Temperature not found. ' +
-                                                             'Select the sample temperature field:',
-                                                             ws.run().keys(), False)
+                                                             'Select the sample temperature field or enter a value:',
```

Comment:
My only minor comment here is that I think we should mention the units that we expect the temperature to be in, i.e.
"Select the sample temperature field or enter a value in Kelvin" (assuming it is in Kelvin)